packages/cli/src/modules/insights/__tests__/insights-compaction.service.test.ts

```diff
@@ -594,5 +602,54 @@ if (dbType === 'sqlite' && !globalConfig.database.sqlite.poolSize) {
 				);
 			});
 		});
+
+		describe('Avoid deadlock error', () => {
+			let defaultBatchSize: number;
+			beforeAll(() => {
+				// Store the original config value
+				const insightsConfig = Container.get(InsightsConfig);
+				defaultBatchSize = insightsConfig.compactionBatchSize;
+
+				// Set a smaller batch size to trigger the deadlock error
+				insightsConfig.compactionBatchSize = 3;
+			});
+
+			afterAll(() => {
+				// Reset the config to its original state
+				const insightsConfig = Container.get(InsightsConfig);
+				insightsConfig.compactionBatchSize = defaultBatchSize;
+			});
+
+			test('should not throw deadlock error on concurrent compaction', async () => {
+				// ARRANGE
+				const insightsCompactionService = Container.get(InsightsCompactionService);
+				const insightsByPeriodRepository = Container.get(InsightsByPeriodRepository);
+				const transactionSpy = jest.spyOn(insightsByPeriodRepository.manager, 'transaction');
+				const project = await createTeamProject();
+				const workflow = await createWorkflow({}, project);
+				await createMetadata(workflow);
+
+				// Create test data
+				const promises = [];
+				for (let i = 0; i < 100; i++) {
+					await createCompactedInsightsEvent(workflow, {
+						type: 'success',
+						value: 1,
+						periodUnit: 'hour',
+						periodStart: DateTime.now().minus({ day: 91, hour: i + 1 }),
+					});
+				}
+
+				// ACT
+				for (let i = 0; i < 10; i++) {
+					promises.push(insightsCompactionService.compactHourToDay());
```

Comment:
Not a big deal but I'm wondering why we test this by calling `insightsCompactionService.compactHourToDay` instead of having a test file for the repository that calls `insightsByPeriodRepository.compactSourceDataIntoInsightPeriod` directly?

Reply:
Good idea, done